File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/di/component/ViewModelComponent.java
package com.kalashnyk.denys.samplemvvmapp.di.component;

import com.kalashnyk.denys.samplemvvmapp.di.module.ViewModelModule;
import com.kalashnyk.denys.samplemvvmapp.di.scope.ViewModelScope;
import com.kalashnyk.denys.samplemvvmapp.presentation.activities.detail.DetailActivity;
import com.kalashnyk.denys.samplemvvmapp.presentation.activities.main.MainActivity;

import dagger.Component;

@ViewModelScope
@Component(modules = {ViewModelModule.class}, dependencies = {RepositoryComponent.class})
public interface ViewModelComponent {
    void inject(MainActivity activity);
    void inject(DetailActivity activity);
}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/di/component/ViewModelComponent.kt
package com.kalashnyk.denys.kotlinbindingsample.di.component

import com.kalashnyk.denys.kotlinbindingsample.di.module.ViewModelModule
import com.kalashnyk.denys.kotlinbindingsample.di.scope.ViewModelScope
import com.kalashnyk.denys.kotlinbindingsample.presentation.activities.detail.DetailActivity
import com.kalashnyk.denys.kotlinbindingsample.presentation.activities.main.MainActivity
import dagger.Component

@ViewModelScope
@Component(modules = [ViewModelModule::class], dependencies = [RepositoryComponent::class])
interface ViewModelComponent {
    fun inject(activity: MainActivity)
    fun inject(activity: DetailActivity)
}